include/options.h

```diff
@@ -0,0 +1,293 @@
+  bool customShownameEnabled() const;
+  void setCustomShownameEnabled(bool value);
+
+  //Returns the showname the user may have set in config.ini.
+  QString shownameOnJoin() const;
+  void setShownameOnJoin(QString value);
+
+  // Returns if the user has casing alerts Enabled.
+  bool casingAlertEnabled() const;
+  void setCasingAlertEnabled(bool value);
+
+  // Returns if the user wants to get alerts for the defence role.
+  bool casingDefenceEnabled() const;
+  void setcasingDefenceEnabled(bool value);
+
+  // Same for prosecution.
+  bool casingProsecutionEnabled() const;
+  void setCasingProseuctionEnabled(bool value);
+
+  // Same for judge.
+  bool casingJudgeEnabled() const;
+  void setCasingJudgeEnabled(bool value);
+
+  // Same for juror.
+  bool casingJurorEnabled() const;
+  void setCasingJurorEnabled(bool value);
+
+  // Same for steno.
+  bool casingStenoEnabled() const;
+  void setCasingStenoEnabled(bool value);
+
+  // Same for CM.
+  bool casingCmEnabled() const;
+  void setCasingCmEnabled(bool value);
+
+  // Get the message for the CM for casing alerts.
+  QString casingCanHostCases() const;
+  void setCasingCanHostCases(QString value);
+
+  // Get if text file logging is Enabled
+  bool logToTextFileEnabled() const;
+  void setLogToTextFileEnabled(bool value);
+
+  // Get if demo logging is Enabled
+  bool logToDemoFileEnabled() const;
+  void setLogToDemoFileEnabled(bool value);
+
+  // Get the subtheme from settings
+  QString subTheme() const;
+  void setSubTheme(QString value);
+
+  // Get if the theme is animated
+  bool animatedThemeEnabled() const;
+  void setAnimatedThemeEnabled(bool value);
+
+  // Get the default scaling method
+  QString defaultScalingMode() const;
+  void setDefaultScalingMode(QString value);
+
+  // Get a list of custom mount paths
+  QStringList mountpaths() const;
+  void setMountpaths(QStringList value);
+
+  // Get whether to opt out of player count metrics sent to the master server
+  bool PlayercountOptout() const;
+  void SetPlayerCountOptout(bool value);
+
+  // Get if sfx can be sent to play on idle
+  bool SfxonIdle() const;
+  void setSfxOnIdle(bool value);
+
+  // Whether opening evidence requires a single or double click
+  bool evidenceDoubleClickEdit() const;
+  void setEvidenceDoubleClickEdit(bool value);
+
+  //Supplies an alternative masterserver URL
+  QString alternativeMasterserver() const;
+  void setAlternativeMasterserver(QString value);
+
+  // Language the client loads on start.
+  QString language() const;
+  void setLanguage(QString value);
+
+  // Callwords notify the user when the word/words are used in a game message.
+  QStringList callwords() const;
+  void setCallwords(QStringList value);
+
+  // Clears the configuration file. Essentially restoring it to default.
+  void clearConfig();
+
+  /**
+   * @brief Static access pointer. Workaround for scope limitations and reduce reliance to AOApplication
+   */
+  inline static Options* options;
```

Comment:
this is a singleton class, so you need to follow the C++ best practices for singletons: *book thump*

https://stackoverflow.com/questions/1008019/c-singleton-design-pattern/1008289#1008289

it is not a hard practice to follow, but it is the kind of thing that reminds you that C++ is still a loaded gun even with layers of Qt.